numpyro/infer/hmc_gibbs.py

```diff
@@ -787,7 +789,7 @@ def log_likelihood(params_flat, subsample_indices=None):
                             log_lik[frame.name] += _sum_all_except_at_dim(
                                 site["fn"].log_prob(site["value"]), frame.dim
                             )
-                        else:
+                        elif frame.name in subsample_indices:
```

Comment:
This is where the bug happens. We only want to collect likelihoods at subsample dimensions.